{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ebe8f202-aaad-471b-a669-627665cf96f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import dill\n",
    "infile = open('/home/adufour/work/scenic_omics/stemcells/SCENIC/scplus_obj.pkl', 'rb')\n",
    "scplus_obj = dill.load(infile)\n",
    "infile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3aa04bf1-2098-4899-878e-1d0e79d22350",
   "metadata": {},
   "outputs": [],
   "source": [
    "from scenicplus.utils import format_egrns\n",
    "format_egrns(scplus_obj, eregulons_key = 'eRegulons_importance', TF2G_key = 'TF2G_adj', key_added = 'eRegulon_metadata')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3a11348d-9ac8-4e8e-9d93-d63cf0643505",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Region_signature_name</th>\n",
       "      <th>Gene_signature_name</th>\n",
       "      <th>TF</th>\n",
       "      <th>is_extended</th>\n",
       "      <th>Region</th>\n",
       "      <th>Gene</th>\n",
       "      <th>R2G_importance</th>\n",
       "      <th>R2G_rho</th>\n",
       "      <th>R2G_importance_x_rho</th>\n",
       "      <th>R2G_importance_x_abs_rho</th>\n",
       "      <th>TF2G_importance</th>\n",
       "      <th>TF2G_regulation</th>\n",
       "      <th>TF2G_rho</th>\n",
       "      <th>TF2G_importance_x_abs_rho</th>\n",
       "      <th>TF2G_importance_x_rho</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSSSCG00000000272_+_+_(31r)</td>\n",
       "      <td>ENSSSCG00000000272_+_+_(18g)</td>\n",
       "      <td>ENSSSCG00000000272</td>\n",
       "      <td>False</td>\n",
       "      <td>1:237788971-237789471</td>\n",
       "      <td>ENSSSCG00000036086</td>\n",
       "      <td>0.017435</td>\n",
       "      <td>0.133475</td>\n",
       "      <td>0.002327</td>\n",
       "      <td>0.002327</td>\n",
       "      <td>0.182510</td>\n",
       "      <td>1</td>\n",
       "      <td>0.062649</td>\n",
       "      <td>0.011434</td>\n",
       "      <td>0.011434</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSSSCG00000000272_+_+_(31r)</td>\n",
       "      <td>ENSSSCG00000000272_+_+_(18g)</td>\n",
       "      <td>ENSSSCG00000000272</td>\n",
       "      <td>False</td>\n",
       "      <td>12:36061522-36062022</td>\n",
       "      <td>ENSSSCG00000017670</td>\n",
       "      <td>0.033763</td>\n",
       "      <td>0.083223</td>\n",
       "      <td>0.002810</td>\n",
       "      <td>0.002810</td>\n",
       "      <td>0.172797</td>\n",
       "      <td>1</td>\n",
       "      <td>0.061065</td>\n",
       "      <td>0.010552</td>\n",
       "      <td>0.010552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSSSCG00000000272_+_+_(31r)</td>\n",
       "      <td>ENSSSCG00000000272_+_+_(18g)</td>\n",
       "      <td>ENSSSCG00000000272</td>\n",
       "      <td>False</td>\n",
       "      <td>12:36069143-36069643</td>\n",
       "      <td>ENSSSCG00000017670</td>\n",
       "      <td>0.012958</td>\n",
       "      <td>0.054082</td>\n",
       "      <td>0.000701</td>\n",
       "      <td>0.000701</td>\n",
       "      <td>0.172797</td>\n",
       "      <td>1</td>\n",
       "      <td>0.061065</td>\n",
       "      <td>0.010552</td>\n",
       "      <td>0.010552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSSSCG00000000272_+_+_(31r)</td>\n",
       "      <td>ENSSSCG00000000272_+_+_(18g)</td>\n",
       "      <td>ENSSSCG00000000272</td>\n",
       "      <td>False</td>\n",
       "      <td>12:36069720-36070220</td>\n",
       "      <td>ENSSSCG00000017670</td>\n",
       "      <td>0.021281</td>\n",
       "      <td>0.073116</td>\n",
       "      <td>0.001556</td>\n",
       "      <td>0.001556</td>\n",
       "      <td>0.172797</td>\n",
       "      <td>1</td>\n",
       "      <td>0.061065</td>\n",
       "      <td>0.010552</td>\n",
       "      <td>0.010552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSSSCG00000000272_+_+_(31r)</td>\n",
       "      <td>ENSSSCG00000000272_+_+_(18g)</td>\n",
       "      <td>ENSSSCG00000000272</td>\n",
       "      <td>False</td>\n",
       "      <td>16:55350778-55351278</td>\n",
       "      <td>ENSSSCG00000017014</td>\n",
       "      <td>0.055225</td>\n",
       "      <td>0.160718</td>\n",
       "      <td>0.008876</td>\n",
       "      <td>0.008876</td>\n",
       "      <td>0.247606</td>\n",
       "      <td>1</td>\n",
       "      <td>0.072075</td>\n",
       "      <td>0.017846</td>\n",
       "      <td>0.017846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ENSSSCG00000036033_extended_-_-_(20r)</td>\n",
       "      <td>ENSSSCG00000036033_extended_-_-_(19g)</td>\n",
       "      <td>ENSSSCG00000036033</td>\n",
       "      <td>True</td>\n",
       "      <td>6:106138371-106138871</td>\n",
       "      <td>ENSSSCG00000040397</td>\n",
       "      <td>0.005845</td>\n",
       "      <td>-0.059991</td>\n",
       "      <td>-0.000351</td>\n",
       "      <td>0.000351</td>\n",
       "      <td>0.622918</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.065444</td>\n",
       "      <td>-0.040766</td>\n",
       "      <td>0.040766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ENSSSCG00000036033_extended_-_-_(20r)</td>\n",
       "      <td>ENSSSCG00000036033_extended_-_-_(19g)</td>\n",
       "      <td>ENSSSCG00000036033</td>\n",
       "      <td>True</td>\n",
       "      <td>14:121349228-121349728</td>\n",
       "      <td>ENSSSCG00000010627</td>\n",
       "      <td>0.012293</td>\n",
       "      <td>-0.188994</td>\n",
       "      <td>-0.002323</td>\n",
       "      <td>0.002323</td>\n",
       "      <td>1.034489</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.110379</td>\n",
       "      <td>-0.114186</td>\n",
       "      <td>0.114186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ENSSSCG00000036033_extended_-_-_(20r)</td>\n",
       "      <td>ENSSSCG00000036033_extended_-_-_(19g)</td>\n",
       "      <td>ENSSSCG00000036033</td>\n",
       "      <td>True</td>\n",
       "      <td>2:75510570-75511070</td>\n",
       "      <td>ENSSSCG00000013476</td>\n",
       "      <td>0.009085</td>\n",
       "      <td>-0.058602</td>\n",
       "      <td>-0.000532</td>\n",
       "      <td>0.000532</td>\n",
       "      <td>0.786519</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.068877</td>\n",
       "      <td>-0.054173</td>\n",
       "      <td>0.054173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ENSSSCG00000036033_extended_-_-_(20r)</td>\n",
       "      <td>ENSSSCG00000036033_extended_-_-_(19g)</td>\n",
       "      <td>ENSSSCG00000036033</td>\n",
       "      <td>True</td>\n",
       "      <td>16:21557266-21557766</td>\n",
       "      <td>ENSSSCG00000016835</td>\n",
       "      <td>0.053197</td>\n",
       "      <td>-0.139667</td>\n",
       "      <td>-0.007430</td>\n",
       "      <td>0.007430</td>\n",
       "      <td>1.065438</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.122840</td>\n",
       "      <td>-0.130878</td>\n",
       "      <td>0.130878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ENSSSCG00000036033_extended_-_-_(20r)</td>\n",
       "      <td>ENSSSCG00000036033_extended_-_-_(19g)</td>\n",
       "      <td>ENSSSCG00000036033</td>\n",
       "      <td>True</td>\n",
       "      <td>7:80161753-80162253</td>\n",
       "      <td>ENSSSCG00000002243</td>\n",
       "      <td>0.063513</td>\n",
       "      <td>-0.127726</td>\n",
       "      <td>-0.008112</td>\n",
       "      <td>0.008112</td>\n",
       "      <td>1.667412</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.104818</td>\n",
       "      <td>-0.174775</td>\n",
       "      <td>0.174775</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33865 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Region_signature_name  \\\n",
       "0            ENSSSCG00000000272_+_+_(31r)   \n",
       "1            ENSSSCG00000000272_+_+_(31r)   \n",
       "2            ENSSSCG00000000272_+_+_(31r)   \n",
       "3            ENSSSCG00000000272_+_+_(31r)   \n",
       "4            ENSSSCG00000000272_+_+_(31r)   \n",
       "..                                    ...   \n",
       "15  ENSSSCG00000036033_extended_-_-_(20r)   \n",
       "16  ENSSSCG00000036033_extended_-_-_(20r)   \n",
       "17  ENSSSCG00000036033_extended_-_-_(20r)   \n",
       "18  ENSSSCG00000036033_extended_-_-_(20r)   \n",
       "19  ENSSSCG00000036033_extended_-_-_(20r)   \n",
       "\n",
       "                      Gene_signature_name                  TF is_extended  \\\n",
       "0            ENSSSCG00000000272_+_+_(18g)  ENSSSCG00000000272       False   \n",
       "1            ENSSSCG00000000272_+_+_(18g)  ENSSSCG00000000272       False   \n",
       "2            ENSSSCG00000000272_+_+_(18g)  ENSSSCG00000000272       False   \n",
       "3            ENSSSCG00000000272_+_+_(18g)  ENSSSCG00000000272       False   \n",
       "4            ENSSSCG00000000272_+_+_(18g)  ENSSSCG00000000272       False   \n",
       "..                                    ...                 ...         ...   \n",
       "15  ENSSSCG00000036033_extended_-_-_(19g)  ENSSSCG00000036033        True   \n",
       "16  ENSSSCG00000036033_extended_-_-_(19g)  ENSSSCG00000036033        True   \n",
       "17  ENSSSCG00000036033_extended_-_-_(19g)  ENSSSCG00000036033        True   \n",
       "18  ENSSSCG00000036033_extended_-_-_(19g)  ENSSSCG00000036033        True   \n",
       "19  ENSSSCG00000036033_extended_-_-_(19g)  ENSSSCG00000036033        True   \n",
       "\n",
       "                    Region                Gene  R2G_importance   R2G_rho  \\\n",
       "0    1:237788971-237789471  ENSSSCG00000036086        0.017435  0.133475   \n",
       "1     12:36061522-36062022  ENSSSCG00000017670        0.033763  0.083223   \n",
       "2     12:36069143-36069643  ENSSSCG00000017670        0.012958  0.054082   \n",
       "3     12:36069720-36070220  ENSSSCG00000017670        0.021281  0.073116   \n",
       "4     16:55350778-55351278  ENSSSCG00000017014        0.055225  0.160718   \n",
       "..                     ...                 ...             ...       ...   \n",
       "15   6:106138371-106138871  ENSSSCG00000040397        0.005845 -0.059991   \n",
       "16  14:121349228-121349728  ENSSSCG00000010627        0.012293 -0.188994   \n",
       "17     2:75510570-75511070  ENSSSCG00000013476        0.009085 -0.058602   \n",
       "18    16:21557266-21557766  ENSSSCG00000016835        0.053197 -0.139667   \n",
       "19     7:80161753-80162253  ENSSSCG00000002243        0.063513 -0.127726   \n",
       "\n",
       "    R2G_importance_x_rho  R2G_importance_x_abs_rho  TF2G_importance  \\\n",
       "0               0.002327                  0.002327         0.182510   \n",
       "1               0.002810                  0.002810         0.172797   \n",
       "2               0.000701                  0.000701         0.172797   \n",
       "3               0.001556                  0.001556         0.172797   \n",
       "4               0.008876                  0.008876         0.247606   \n",
       "..                   ...                       ...              ...   \n",
       "15             -0.000351                  0.000351         0.622918   \n",
       "16             -0.002323                  0.002323         1.034489   \n",
       "17             -0.000532                  0.000532         0.786519   \n",
       "18             -0.007430                  0.007430         1.065438   \n",
       "19             -0.008112                  0.008112         1.667412   \n",
       "\n",
       "    TF2G_regulation  TF2G_rho  TF2G_importance_x_abs_rho  \\\n",
       "0                 1  0.062649                   0.011434   \n",
       "1                 1  0.061065                   0.010552   \n",
       "2                 1  0.061065                   0.010552   \n",
       "3                 1  0.061065                   0.010552   \n",
       "4                 1  0.072075                   0.017846   \n",
       "..              ...       ...                        ...   \n",
       "15               -1 -0.065444                  -0.040766   \n",
       "16               -1 -0.110379                  -0.114186   \n",
       "17               -1 -0.068877                  -0.054173   \n",
       "18               -1 -0.122840                  -0.130878   \n",
       "19               -1 -0.104818                  -0.174775   \n",
       "\n",
       "    TF2G_importance_x_rho  \n",
       "0                0.011434  \n",
       "1                0.010552  \n",
       "2                0.010552  \n",
       "3                0.010552  \n",
       "4                0.017846  \n",
       "..                    ...  \n",
       "15               0.040766  \n",
       "16               0.114186  \n",
       "17               0.054173  \n",
       "18               0.130878  \n",
       "19               0.174775  \n",
       "\n",
       "[33865 rows x 15 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scplus_obj.uns['eRegulon_metadata']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c9bd0501-90a4-4660-ab19-9814789f5bc1",
   "metadata": {},
   "outputs": [],
   "source": [
    "def gene_convert(gene_list):\n",
    "    from gprofiler import GProfiler\n",
    "    import numpy as np\n",
    "    gp = GProfiler(return_dataframe=True)\n",
    "    x = np.array(gene_list)\n",
    "    gene_list_convert = gp.convert(organism='sscrofa', query=np.unique(x).tolist())\n",
    "    gene_list_convert['name'] = np.where(gene_list_convert['name'] == 'None', gene_list_convert['incoming'], gene_list_convert['name'])\n",
    "    return gene_list_convert"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0dc9be4b-20b3-431c-97f2-87d9a4216ee2",
   "metadata": {},
   "outputs": [],
   "source": [
    "list_tf = gene_convert(scplus_obj.uns['eRegulon_metadata']['TF'].tolist())\n",
    "list_gene = gene_convert(scplus_obj.uns['eRegulon_metadata']['Gene'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "6c61cbb1-eaba-4bfd-b754-0834d9517c28",
   "metadata": {},
   "outputs": [],
   "source": [
    "scplus_obj.uns['eRegulon_metadata']['TF'] = scplus_obj.uns['eRegulon_metadata']['TF'].replace(list_tf['incoming'].tolist(), list_tf['name'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "55588d3b-6685-4475-a9d7-1477d0ab84f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "scplus_obj.uns['eRegulon_metadata']['Gene'] = scplus_obj.uns['eRegulon_metadata']['Gene'].replace(list_gene['incoming'].tolist(), list_gene['name'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d948c08d-fe97-44ba-90a5-e1fb181bfd5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def gene_convert_part(gene_list):\n",
    "    from gprofiler import GProfiler\n",
    "    import numpy as np\n",
    "    gp = GProfiler(return_dataframe=True)\n",
    "    x = np.array(gene_list)\n",
    "    x = np.unique(x).tolist()\n",
    "    part_a = [item.split('_', 1)[0] for item in x]\n",
    "    part_convert = gp.convert(organism='sscrofa', query=part_a)\n",
    "    part_convert['name'] = np.where(part_convert['name'] == 'None', part_convert['incoming'], part_convert['name'])\n",
    "    part_b = [item.split('_', 1)[1] for item in x]\n",
    "    region_list = [f\"{itema}_{itemb}\" for itema, itemb in zip(part_convert['name'], part_b)]\n",
    "    return x, region_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "0ad45863-7b7f-428f-8cc3-51b7941d11ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "source, replacement = gene_convert_part(scplus_obj.uns['eRegulon_metadata']['Region_signature_name'].tolist())\n",
    "scplus_obj.uns['eRegulon_metadata']['Region_signature_name'] = scplus_obj.uns['eRegulon_metadata']['Region_signature_name'].replace(source, replacement)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "2c74deaf-e6f7-4994-a1b1-cd3487473996",
   "metadata": {},
   "outputs": [],
   "source": [
    "source, replacement = gene_convert_part(scplus_obj.uns['eRegulon_metadata']['Gene_signature_name'].tolist())\n",
    "scplus_obj.uns['eRegulon_metadata']['Gene_signature_name'] = scplus_obj.uns['eRegulon_metadata']['Gene_signature_name'].replace(source, replacement)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "244e4c7b-035a-4339-b58e-5207963e93aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Region_signature_name</th>\n",
       "      <th>Gene_signature_name</th>\n",
       "      <th>TF</th>\n",
       "      <th>is_extended</th>\n",
       "      <th>Region</th>\n",
       "      <th>Gene</th>\n",
       "      <th>R2G_importance</th>\n",
       "      <th>R2G_rho</th>\n",
       "      <th>R2G_importance_x_rho</th>\n",
       "      <th>R2G_importance_x_abs_rho</th>\n",
       "      <th>TF2G_importance</th>\n",
       "      <th>TF2G_regulation</th>\n",
       "      <th>TF2G_rho</th>\n",
       "      <th>TF2G_importance_x_abs_rho</th>\n",
       "      <th>TF2G_importance_x_rho</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SP1_+_+_(31r)</td>\n",
       "      <td>SP1_+_+_(18g)</td>\n",
       "      <td>SP1</td>\n",
       "      <td>False</td>\n",
       "      <td>1:237788971-237789471</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>0.017435</td>\n",
       "      <td>0.133475</td>\n",
       "      <td>0.002327</td>\n",
       "      <td>0.002327</td>\n",
       "      <td>0.182510</td>\n",
       "      <td>1</td>\n",
       "      <td>0.062649</td>\n",
       "      <td>0.011434</td>\n",
       "      <td>0.011434</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SP1_+_+_(31r)</td>\n",
       "      <td>SP1_+_+_(18g)</td>\n",
       "      <td>SP1</td>\n",
       "      <td>False</td>\n",
       "      <td>12:36061522-36062022</td>\n",
       "      <td>RPS6KB1</td>\n",
       "      <td>0.033763</td>\n",
       "      <td>0.083223</td>\n",
       "      <td>0.002810</td>\n",
       "      <td>0.002810</td>\n",
       "      <td>0.172797</td>\n",
       "      <td>1</td>\n",
       "      <td>0.061065</td>\n",
       "      <td>0.010552</td>\n",
       "      <td>0.010552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SP1_+_+_(31r)</td>\n",
       "      <td>SP1_+_+_(18g)</td>\n",
       "      <td>SP1</td>\n",
       "      <td>False</td>\n",
       "      <td>12:36069143-36069643</td>\n",
       "      <td>RPS6KB1</td>\n",
       "      <td>0.012958</td>\n",
       "      <td>0.054082</td>\n",
       "      <td>0.000701</td>\n",
       "      <td>0.000701</td>\n",
       "      <td>0.172797</td>\n",
       "      <td>1</td>\n",
       "      <td>0.061065</td>\n",
       "      <td>0.010552</td>\n",
       "      <td>0.010552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SP1_+_+_(31r)</td>\n",
       "      <td>SP1_+_+_(18g)</td>\n",
       "      <td>SP1</td>\n",
       "      <td>False</td>\n",
       "      <td>12:36069720-36070220</td>\n",
       "      <td>RPS6KB1</td>\n",
       "      <td>0.021281</td>\n",
       "      <td>0.073116</td>\n",
       "      <td>0.001556</td>\n",
       "      <td>0.001556</td>\n",
       "      <td>0.172797</td>\n",
       "      <td>1</td>\n",
       "      <td>0.061065</td>\n",
       "      <td>0.010552</td>\n",
       "      <td>0.010552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SP1_+_+_(31r)</td>\n",
       "      <td>SP1_+_+_(18g)</td>\n",
       "      <td>SP1</td>\n",
       "      <td>False</td>\n",
       "      <td>16:55350778-55351278</td>\n",
       "      <td>PANK3</td>\n",
       "      <td>0.055225</td>\n",
       "      <td>0.160718</td>\n",
       "      <td>0.008876</td>\n",
       "      <td>0.008876</td>\n",
       "      <td>0.247606</td>\n",
       "      <td>1</td>\n",
       "      <td>0.072075</td>\n",
       "      <td>0.017846</td>\n",
       "      <td>0.017846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>THRB_extended_-_-_(20r)</td>\n",
       "      <td>THRB_extended_-_-_(19g)</td>\n",
       "      <td>THRB</td>\n",
       "      <td>True</td>\n",
       "      <td>6:106138371-106138871</td>\n",
       "      <td>COLEC12</td>\n",
       "      <td>0.005845</td>\n",
       "      <td>-0.059991</td>\n",
       "      <td>-0.000351</td>\n",
       "      <td>0.000351</td>\n",
       "      <td>0.622918</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.065444</td>\n",
       "      <td>-0.040766</td>\n",
       "      <td>0.040766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>THRB_extended_-_-_(20r)</td>\n",
       "      <td>THRB_extended_-_-_(19g)</td>\n",
       "      <td>THRB</td>\n",
       "      <td>True</td>\n",
       "      <td>14:121349228-121349728</td>\n",
       "      <td>PDCD4</td>\n",
       "      <td>0.012293</td>\n",
       "      <td>-0.188994</td>\n",
       "      <td>-0.002323</td>\n",
       "      <td>0.002323</td>\n",
       "      <td>1.034489</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.110379</td>\n",
       "      <td>-0.114186</td>\n",
       "      <td>0.114186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>THRB_extended_-_-_(20r)</td>\n",
       "      <td>THRB_extended_-_-_(19g)</td>\n",
       "      <td>THRB</td>\n",
       "      <td>True</td>\n",
       "      <td>2:75510570-75511070</td>\n",
       "      <td>CELF5</td>\n",
       "      <td>0.009085</td>\n",
       "      <td>-0.058602</td>\n",
       "      <td>-0.000532</td>\n",
       "      <td>0.000532</td>\n",
       "      <td>0.786519</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.068877</td>\n",
       "      <td>-0.054173</td>\n",
       "      <td>0.054173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>THRB_extended_-_-_(20r)</td>\n",
       "      <td>THRB_extended_-_-_(19g)</td>\n",
       "      <td>THRB</td>\n",
       "      <td>True</td>\n",
       "      <td>16:21557266-21557766</td>\n",
       "      <td>SKP2</td>\n",
       "      <td>0.053197</td>\n",
       "      <td>-0.139667</td>\n",
       "      <td>-0.007430</td>\n",
       "      <td>0.007430</td>\n",
       "      <td>1.065438</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.122840</td>\n",
       "      <td>-0.130878</td>\n",
       "      <td>0.130878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>THRB_extended_-_-_(20r)</td>\n",
       "      <td>THRB_extended_-_-_(19g)</td>\n",
       "      <td>THRB</td>\n",
       "      <td>True</td>\n",
       "      <td>7:80161753-80162253</td>\n",
       "      <td>SLC12A6</td>\n",
       "      <td>0.063513</td>\n",
       "      <td>-0.127726</td>\n",
       "      <td>-0.008112</td>\n",
       "      <td>0.008112</td>\n",
       "      <td>1.667412</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.104818</td>\n",
       "      <td>-0.174775</td>\n",
       "      <td>0.174775</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33865 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Region_signature_name      Gene_signature_name    TF is_extended  \\\n",
       "0             SP1_+_+_(31r)            SP1_+_+_(18g)   SP1       False   \n",
       "1             SP1_+_+_(31r)            SP1_+_+_(18g)   SP1       False   \n",
       "2             SP1_+_+_(31r)            SP1_+_+_(18g)   SP1       False   \n",
       "3             SP1_+_+_(31r)            SP1_+_+_(18g)   SP1       False   \n",
       "4             SP1_+_+_(31r)            SP1_+_+_(18g)   SP1       False   \n",
       "..                      ...                      ...   ...         ...   \n",
       "15  THRB_extended_-_-_(20r)  THRB_extended_-_-_(19g)  THRB        True   \n",
       "16  THRB_extended_-_-_(20r)  THRB_extended_-_-_(19g)  THRB        True   \n",
       "17  THRB_extended_-_-_(20r)  THRB_extended_-_-_(19g)  THRB        True   \n",
       "18  THRB_extended_-_-_(20r)  THRB_extended_-_-_(19g)  THRB        True   \n",
       "19  THRB_extended_-_-_(20r)  THRB_extended_-_-_(19g)  THRB        True   \n",
       "\n",
       "                    Region     Gene  R2G_importance   R2G_rho  \\\n",
       "0    1:237788971-237789471     PAX5        0.017435  0.133475   \n",
       "1     12:36061522-36062022  RPS6KB1        0.033763  0.083223   \n",
       "2     12:36069143-36069643  RPS6KB1        0.012958  0.054082   \n",
       "3     12:36069720-36070220  RPS6KB1        0.021281  0.073116   \n",
       "4     16:55350778-55351278    PANK3        0.055225  0.160718   \n",
       "..                     ...      ...             ...       ...   \n",
       "15   6:106138371-106138871  COLEC12        0.005845 -0.059991   \n",
       "16  14:121349228-121349728    PDCD4        0.012293 -0.188994   \n",
       "17     2:75510570-75511070    CELF5        0.009085 -0.058602   \n",
       "18    16:21557266-21557766     SKP2        0.053197 -0.139667   \n",
       "19     7:80161753-80162253  SLC12A6        0.063513 -0.127726   \n",
       "\n",
       "    R2G_importance_x_rho  R2G_importance_x_abs_rho  TF2G_importance  \\\n",
       "0               0.002327                  0.002327         0.182510   \n",
       "1               0.002810                  0.002810         0.172797   \n",
       "2               0.000701                  0.000701         0.172797   \n",
       "3               0.001556                  0.001556         0.172797   \n",
       "4               0.008876                  0.008876         0.247606   \n",
       "..                   ...                       ...              ...   \n",
       "15             -0.000351                  0.000351         0.622918   \n",
       "16             -0.002323                  0.002323         1.034489   \n",
       "17             -0.000532                  0.000532         0.786519   \n",
       "18             -0.007430                  0.007430         1.065438   \n",
       "19             -0.008112                  0.008112         1.667412   \n",
       "\n",
       "    TF2G_regulation  TF2G_rho  TF2G_importance_x_abs_rho  \\\n",
       "0                 1  0.062649                   0.011434   \n",
       "1                 1  0.061065                   0.010552   \n",
       "2                 1  0.061065                   0.010552   \n",
       "3                 1  0.061065                   0.010552   \n",
       "4                 1  0.072075                   0.017846   \n",
       "..              ...       ...                        ...   \n",
       "15               -1 -0.065444                  -0.040766   \n",
       "16               -1 -0.110379                  -0.114186   \n",
       "17               -1 -0.068877                  -0.054173   \n",
       "18               -1 -0.122840                  -0.130878   \n",
       "19               -1 -0.104818                  -0.174775   \n",
       "\n",
       "    TF2G_importance_x_rho  \n",
       "0                0.011434  \n",
       "1                0.010552  \n",
       "2                0.010552  \n",
       "3                0.010552  \n",
       "4                0.017846  \n",
       "..                    ...  \n",
       "15               0.040766  \n",
       "16               0.114186  \n",
       "17               0.054173  \n",
       "18               0.130878  \n",
       "19               0.174775  \n",
       "\n",
       "[33865 rows x 15 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scplus_obj.uns['eRegulon_metadata']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "bbd5b0af-49f6-480d-a4f8-5d4396ced8a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "list_gene = gene_convert(scplus_obj.metadata_genes['gene_ids'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "b624cb97-5d5f-48b5-acdd-847af87f1bf8",
   "metadata": {},
   "outputs": [],
   "source": [
    "scplus_obj.metadata_genes['gene_ids'] = scplus_obj.metadata_genes['gene_ids'].replace(list_gene['incoming'].tolist(), list_gene['name'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "7abc04c4-2c94-463f-a92a-b346e4d1ec16",
   "metadata": {},
   "outputs": [],
   "source": [
    "from numpy import copy\n",
    "newArray = copy(scplus_obj.metadata_genes.index.values)\n",
    "for k, v in zip(list_gene['incoming'].tolist(), list_gene['name'].tolist()): scplus_obj.metadata_genes.index.values[scplus_obj.metadata_genes.index.values==k] = v"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e1b4d279-15ae-4588-8f75-43988b236d6a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_ids</th>\n",
       "      <th>feature_types</th>\n",
       "      <th>genome</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_ids</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSSSCG00000048769</th>\n",
       "      <td>ENSSSCG00000048769</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PSMB1</th>\n",
       "      <td>PSMB1</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DLL1</th>\n",
       "      <td>DLL1</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DYNLT2</th>\n",
       "      <td>DYNLT2</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PHF10</th>\n",
       "      <td>PHF10</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSSSCG00000036928</th>\n",
       "      <td>ENSSSCG00000036928</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSSSCG00000040795</th>\n",
       "      <td>ENSSSCG00000040795</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSSSCG00000023819</th>\n",
       "      <td>ENSSSCG00000023819</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRY</th>\n",
       "      <td>SRY</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSSSCG00000034141</th>\n",
       "      <td>ENSSSCG00000034141</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>35670 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              gene_ids    feature_types  \\\n",
       "gene_ids                                                  \n",
       "ENSSSCG00000048769  ENSSSCG00000048769  Gene Expression   \n",
       "PSMB1                            PSMB1  Gene Expression   \n",
       "DLL1                              DLL1  Gene Expression   \n",
       "DYNLT2                          DYNLT2  Gene Expression   \n",
       "PHF10                            PHF10  Gene Expression   \n",
       "...                                ...              ...   \n",
       "ENSSSCG00000036928  ENSSSCG00000036928  Gene Expression   \n",
       "ENSSSCG00000040795  ENSSSCG00000040795  Gene Expression   \n",
       "ENSSSCG00000023819  ENSSSCG00000023819  Gene Expression   \n",
       "SRY                                SRY  Gene Expression   \n",
       "ENSSSCG00000034141  ENSSSCG00000034141  Gene Expression   \n",
       "\n",
       "                                           genome  \n",
       "gene_ids                                           \n",
       "ENSSSCG00000048769  sus_scrofa_11_ensembl_arc_108  \n",
       "PSMB1               sus_scrofa_11_ensembl_arc_108  \n",
       "DLL1                sus_scrofa_11_ensembl_arc_108  \n",
       "DYNLT2              sus_scrofa_11_ensembl_arc_108  \n",
       "PHF10               sus_scrofa_11_ensembl_arc_108  \n",
       "...                                           ...  \n",
       "ENSSSCG00000036928  sus_scrofa_11_ensembl_arc_108  \n",
       "ENSSSCG00000040795  sus_scrofa_11_ensembl_arc_108  \n",
       "ENSSSCG00000023819  sus_scrofa_11_ensembl_arc_108  \n",
       "SRY                 sus_scrofa_11_ensembl_arc_108  \n",
       "ENSSSCG00000034141  sus_scrofa_11_ensembl_arc_108  \n",
       "\n",
       "[35670 rows x 3 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scplus_obj.metadata_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "05f0041d-d0b6-48a1-919f-9953d36a0fce",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Format eRegulons\n",
    "from scenicplus.eregulon_enrichment import *\n",
    "get_eRegulons_as_signatures(scplus_obj, eRegulon_metadata_key='eRegulon_metadata', key_added='eRegulon_signatures')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "905708e6-e801-4324-8010-8ecc2b099c4f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 314/314 [00:35<00:00,  8.78it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10.216791987419128\n"
     ]
    }
   ],
   "source": [
    "## Score chromatin layer\n",
    "# Region based raking\n",
    "from scenicplus.cistromes import *\n",
    "import time\n",
    "start_time = time.time()\n",
    "region_ranking = make_rankings(scplus_obj, target='region')\n",
    "# Score region regulons\n",
    "score_eRegulons(scplus_obj,\n",
    "                ranking = region_ranking,\n",
    "                eRegulon_signatures_key = 'eRegulon_signatures',\n",
    "                key_added = 'eRegulon_AUC',\n",
    "                enrichment_type= 'region',\n",
    "                auc_threshold = 0.05,\n",
    "                normalize = False,\n",
    "                n_cpu = 1)\n",
    "time = time.time()-start_time\n",
    "print(time/60)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f83d21c4-ab8e-46f9-8c7c-2371a1763cfd",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 314/314 [00:20<00:00, 15.37it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.3796157240867615\n"
     ]
    }
   ],
   "source": [
    "## Score transcriptome layer\n",
    "# Gene based raking\n",
    "from scenicplus.cistromes import *\n",
    "import time\n",
    "start_time = time.time()\n",
    "gene_ranking = make_rankings(scplus_obj, target='gene')\n",
    "# Score gene regulons\n",
    "score_eRegulons(scplus_obj,\n",
    "                gene_ranking,\n",
    "                eRegulon_signatures_key = 'eRegulon_signatures',\n",
    "                key_added = 'eRegulon_AUC',\n",
    "                enrichment_type = 'gene',\n",
    "                auc_threshold = 0.05,\n",
    "                normalize= False,\n",
    "                n_cpu = 1)\n",
    "time = time.time()-start_time\n",
    "print(time/60)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "db0d24f6-7365-48a7-ae7e-786418130060",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.43580494324366253\n"
     ]
    }
   ],
   "source": [
    "# Generate pseudobulks\n",
    "import time\n",
    "start_time = time.time()\n",
    "generate_pseudobulks(scplus_obj,\n",
    "                         variable = 'GEX_Clusters_Time',\n",
    "                         auc_key = 'eRegulon_AUC',\n",
    "                         signature_key = 'Gene_based',\n",
    "                         nr_cells = 1,\n",
    "                         nr_pseudobulks = 100,\n",
    "                         seed=555)\n",
    "generate_pseudobulks(scplus_obj,\n",
    "                         variable = 'GEX_Clusters_Time',\n",
    "                         auc_key = 'eRegulon_AUC',\n",
    "                         signature_key = 'Region_based',\n",
    "                         nr_cells = 1,\n",
    "                         nr_pseudobulks = 100,\n",
    "                         seed=555)\n",
    "time = time.time()-start_time\n",
    "print(time/60)"
   ]
  },
  {
   "cell_type": "raw",
   "id": "348963bf-f5d2-4faa-b32e-57521181f232",
   "metadata": {},
   "source": [
    "%matplotlib inline\n",
    "import seaborn as sns\n",
    "sns.set_style(\"white\")\n",
    "colors = [\"#E9842C\",\"#F8766D\", \"#BC9D00\", \"#00C0B4\", \"#9CA700\", \"#6FB000\", \"#00B813\", \"#00BD61\", \"#00C08E\", \"#00BDD4\",\n",
    "           \"#00A7FF\", \"#7F96FF\", \"#E26EF7\", \"#FF62BF\", \"#D69100\", \"#BC81FF\"]\n",
    "categories = sorted(set(scplus_obj.metadata_cell['GEX_Clusters']))\n",
    "color_dict = dict(zip(categories, colors[0:len(categories)]))\n",
    "prune_plot(scplus_obj,\n",
    "           'NANOG_+_+',\n",
    "           pseudobulk_variable = 'GEX_Clusters',\n",
    "           show_dot_plot = True,\n",
    "           show_line_plot = False,\n",
    "           color_dict = color_dict,\n",
    "           use_pseudobulk = True,\n",
    "           auc_key = 'eRegulon_AUC',\n",
    "           signature_key = 'Region_based',\n",
    "           seed=555)"
   ]
  },
  {
   "cell_type": "raw",
   "id": "9541bf7a-19c4-4325-ace7-53a56768273c",
   "metadata": {},
   "source": [
    "# Gene based\n",
    "%matplotlib inline\n",
    "sns.set_style(\"white\")\n",
    "colors = [\"#E9842C\",\"#F8766D\", \"#BC9D00\", \"#00C0B4\", \"#9CA700\", \"#6FB000\", \"#00B813\", \"#00BD61\", \"#00C08E\", \"#00BDD4\",\n",
    "           \"#00A7FF\", \"#7F96FF\", \"#E26EF7\", \"#FF62BF\", \"#D69100\", \"#BC81FF\"]\n",
    "categories = sorted(set(scplus_obj.metadata_cell['GEX_Clusters']))\n",
    "color_dict = dict(zip(categories, colors[0:len(categories)]))\n",
    "prune_plot(scplus_obj,\n",
    "           'NANOG_+_+',\n",
    "           pseudobulk_variable = 'GEX_Clusters',\n",
    "           show_dot_plot = True,\n",
    "           show_line_plot = False,\n",
    "           color_dict = color_dict,\n",
    "           use_pseudobulk = True,\n",
    "           auc_key = 'eRegulon_AUC',\n",
    "           signature_key = 'Gene_based',\n",
    "           seed=555)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "3711dbaa-9169-4135-b4e9-7c319ef4626d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correlation between region based regulons and gene based regulons\n",
    "import pandas\n",
    "df1 = scplus_obj.uns['eRegulon_AUC']['Gene_based'].copy()\n",
    "df2 = scplus_obj.uns['eRegulon_AUC']['Region_based'].copy()\n",
    "df1.columns = [x.split('_(')[0] for x in df1.columns]\n",
    "df2.columns = [x.split('_(')[0] for x in df2.columns]\n",
    "correlations = df1.corrwith(df2, axis = 0)\n",
    "correlations = correlations[abs(correlations) > 0.2]\n",
    "# Kepp only R2G +\n",
    "keep = [x for x in correlations.index if '+_+' in x] + [x for x in correlations.index if '-_+' in x]\n",
    "# Keep extended if not direct\n",
    "extended = [x for x in keep if 'extended' in x]\n",
    "direct = [x for x in keep if not 'extended' in x]\n",
    "keep_extended = [x for x in extended if not x.replace('extended_', '') in direct]\n",
    "keep = direct + keep_extended\n",
    "# Keep regulons with more than 10 genes\n",
    "keep_gene = [x for x in scplus_obj.uns['eRegulon_AUC']['Gene_based'].columns if x.split('_(')[0] in keep]\n",
    "keep_gene = [x for x in keep_gene if (int(x.split('_(')[1].replace('g)', '')) > 10)]\n",
    "keep_all = [x.split('_(')[0] for x in keep_gene]\n",
    "keep_region = [x for x in scplus_obj.uns['eRegulon_AUC']['Region_based'].columns if x.split('_(')[0] in keep]\n",
    "scplus_obj.uns['selected_eRegulons'] = {}\n",
    "scplus_obj.uns['selected_eRegulons']['Gene_based'] = keep_gene\n",
    "scplus_obj.uns['selected_eRegulons']['Region_based'] = keep_region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "5bd0bd80-0d4d-49c8-903a-7a74a93fe31f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import dill as dill"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "990152d8-d6ce-4ef7-a024-cee4066b5631",
   "metadata": {},
   "outputs": [],
   "source": [
    "dill.dump(scplus_obj, open('/home/adufour/work/scenic_omics/stemcells/SCENIC/scplus_obj.pkl', 'wb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "43cdf888-4bb0-4e1e-93ff-b819d924fab0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:scenicplus2]",
   "language": "python",
   "name": "conda-env-scenicplus2-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.17"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
