{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ebe8f202-aaad-471b-a669-627665cf96f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import dill\n",
    "infile = open('/home/adufour/work/scenic_omics/embryos/SCENIC/scplus_obj.pkl', 'rb')\n",
    "scplus_obj = dill.load(infile)\n",
    "infile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3aa04bf1-2098-4899-878e-1d0e79d22350",
   "metadata": {},
   "outputs": [],
   "source": [
    "from scenicplus.utils import format_egrns\n",
    "format_egrns(scplus_obj, eregulons_key = 'eRegulons_importance', TF2G_key = 'TF2G_adj', key_added = 'eRegulon_metadata')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3a11348d-9ac8-4e8e-9d93-d63cf0643505",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Region_signature_name</th>\n",
       "      <th>Gene_signature_name</th>\n",
       "      <th>TF</th>\n",
       "      <th>is_extended</th>\n",
       "      <th>Region</th>\n",
       "      <th>Gene</th>\n",
       "      <th>R2G_importance</th>\n",
       "      <th>R2G_rho</th>\n",
       "      <th>R2G_importance_x_rho</th>\n",
       "      <th>R2G_importance_x_abs_rho</th>\n",
       "      <th>TF2G_importance</th>\n",
       "      <th>TF2G_regulation</th>\n",
       "      <th>TF2G_rho</th>\n",
       "      <th>TF2G_importance_x_abs_rho</th>\n",
       "      <th>TF2G_importance_x_rho</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSSSCG00000000272_+_+_(174r)</td>\n",
       "      <td>ENSSSCG00000000272_+_+_(72g)</td>\n",
       "      <td>ENSSSCG00000000272</td>\n",
       "      <td>False</td>\n",
       "      <td>1:107949712-107950212</td>\n",
       "      <td>ENSSSCG00000004558</td>\n",
       "      <td>0.021216</td>\n",
       "      <td>0.132060</td>\n",
       "      <td>0.002802</td>\n",
       "      <td>0.002802</td>\n",
       "      <td>0.877556</td>\n",
       "      <td>1</td>\n",
       "      <td>0.166144</td>\n",
       "      <td>0.145801</td>\n",
       "      <td>0.145801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSSSCG00000000272_+_+_(174r)</td>\n",
       "      <td>ENSSSCG00000000272_+_+_(72g)</td>\n",
       "      <td>ENSSSCG00000000272</td>\n",
       "      <td>False</td>\n",
       "      <td>1:107659273-107659773</td>\n",
       "      <td>ENSSSCG00000004558</td>\n",
       "      <td>0.008690</td>\n",
       "      <td>0.082422</td>\n",
       "      <td>0.000716</td>\n",
       "      <td>0.000716</td>\n",
       "      <td>0.877556</td>\n",
       "      <td>1</td>\n",
       "      <td>0.166144</td>\n",
       "      <td>0.145801</td>\n",
       "      <td>0.145801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSSSCG00000000272_+_+_(174r)</td>\n",
       "      <td>ENSSSCG00000000272_+_+_(72g)</td>\n",
       "      <td>ENSSSCG00000000272</td>\n",
       "      <td>False</td>\n",
       "      <td>1:107689786-107690286</td>\n",
       "      <td>ENSSSCG00000004558</td>\n",
       "      <td>0.021383</td>\n",
       "      <td>0.123316</td>\n",
       "      <td>0.002637</td>\n",
       "      <td>0.002637</td>\n",
       "      <td>0.877556</td>\n",
       "      <td>1</td>\n",
       "      <td>0.166144</td>\n",
       "      <td>0.145801</td>\n",
       "      <td>0.145801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSSSCG00000000272_+_+_(174r)</td>\n",
       "      <td>ENSSSCG00000000272_+_+_(72g)</td>\n",
       "      <td>ENSSSCG00000000272</td>\n",
       "      <td>False</td>\n",
       "      <td>1:107886380-107886880</td>\n",
       "      <td>ENSSSCG00000004558</td>\n",
       "      <td>0.056425</td>\n",
       "      <td>0.161613</td>\n",
       "      <td>0.009119</td>\n",
       "      <td>0.009119</td>\n",
       "      <td>0.877556</td>\n",
       "      <td>1</td>\n",
       "      <td>0.166144</td>\n",
       "      <td>0.145801</td>\n",
       "      <td>0.145801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSSSCG00000000272_+_+_(174r)</td>\n",
       "      <td>ENSSSCG00000000272_+_+_(72g)</td>\n",
       "      <td>ENSSSCG00000000272</td>\n",
       "      <td>False</td>\n",
       "      <td>1:107949165-107949665</td>\n",
       "      <td>ENSSSCG00000004558</td>\n",
       "      <td>0.010226</td>\n",
       "      <td>0.100105</td>\n",
       "      <td>0.001024</td>\n",
       "      <td>0.001024</td>\n",
       "      <td>0.877556</td>\n",
       "      <td>1</td>\n",
       "      <td>0.166144</td>\n",
       "      <td>0.145801</td>\n",
       "      <td>0.145801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ENSSSCG00000037647_extended_-_-_(14r)</td>\n",
       "      <td>ENSSSCG00000037647_extended_-_-_(12g)</td>\n",
       "      <td>ENSSSCG00000037647</td>\n",
       "      <td>True</td>\n",
       "      <td>14:88473878-88474378</td>\n",
       "      <td>ENSSSCG00000010370</td>\n",
       "      <td>0.014041</td>\n",
       "      <td>-0.143928</td>\n",
       "      <td>-0.002021</td>\n",
       "      <td>0.002021</td>\n",
       "      <td>0.678860</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.162168</td>\n",
       "      <td>-0.110089</td>\n",
       "      <td>0.110089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ENSSSCG00000037647_extended_-_-_(14r)</td>\n",
       "      <td>ENSSSCG00000037647_extended_-_-_(12g)</td>\n",
       "      <td>ENSSSCG00000037647</td>\n",
       "      <td>True</td>\n",
       "      <td>5:32970064-32970564</td>\n",
       "      <td>ENSSSCG00000032595</td>\n",
       "      <td>0.016981</td>\n",
       "      <td>-0.109311</td>\n",
       "      <td>-0.001856</td>\n",
       "      <td>0.001856</td>\n",
       "      <td>0.506837</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.149248</td>\n",
       "      <td>-0.075644</td>\n",
       "      <td>0.075644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ENSSSCG00000037647_extended_-_-_(14r)</td>\n",
       "      <td>ENSSSCG00000037647_extended_-_-_(12g)</td>\n",
       "      <td>ENSSSCG00000037647</td>\n",
       "      <td>True</td>\n",
       "      <td>5:32981053-32981553</td>\n",
       "      <td>ENSSSCG00000032595</td>\n",
       "      <td>0.030084</td>\n",
       "      <td>-0.226317</td>\n",
       "      <td>-0.006809</td>\n",
       "      <td>0.006809</td>\n",
       "      <td>0.506837</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.149248</td>\n",
       "      <td>-0.075644</td>\n",
       "      <td>0.075644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ENSSSCG00000037647_extended_-_-_(14r)</td>\n",
       "      <td>ENSSSCG00000037647_extended_-_-_(12g)</td>\n",
       "      <td>ENSSSCG00000037647</td>\n",
       "      <td>True</td>\n",
       "      <td>1:163790120-163790620</td>\n",
       "      <td>ENSSSCG00000044382</td>\n",
       "      <td>0.033541</td>\n",
       "      <td>-0.113878</td>\n",
       "      <td>-0.003820</td>\n",
       "      <td>0.003820</td>\n",
       "      <td>0.805091</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.107096</td>\n",
       "      <td>-0.086222</td>\n",
       "      <td>0.086222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ENSSSCG00000037647_extended_-_-_(14r)</td>\n",
       "      <td>ENSSSCG00000037647_extended_-_-_(12g)</td>\n",
       "      <td>ENSSSCG00000037647</td>\n",
       "      <td>True</td>\n",
       "      <td>12:26842989-26843489</td>\n",
       "      <td>ENSSSCG00000017566</td>\n",
       "      <td>0.031242</td>\n",
       "      <td>-0.393878</td>\n",
       "      <td>-0.012305</td>\n",
       "      <td>0.012305</td>\n",
       "      <td>0.380228</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.207675</td>\n",
       "      <td>-0.078964</td>\n",
       "      <td>0.078964</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41413 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Region_signature_name  \\\n",
       "0           ENSSSCG00000000272_+_+_(174r)   \n",
       "1           ENSSSCG00000000272_+_+_(174r)   \n",
       "2           ENSSSCG00000000272_+_+_(174r)   \n",
       "3           ENSSSCG00000000272_+_+_(174r)   \n",
       "4           ENSSSCG00000000272_+_+_(174r)   \n",
       "..                                    ...   \n",
       "15  ENSSSCG00000037647_extended_-_-_(14r)   \n",
       "16  ENSSSCG00000037647_extended_-_-_(14r)   \n",
       "17  ENSSSCG00000037647_extended_-_-_(14r)   \n",
       "18  ENSSSCG00000037647_extended_-_-_(14r)   \n",
       "19  ENSSSCG00000037647_extended_-_-_(14r)   \n",
       "\n",
       "                      Gene_signature_name                  TF is_extended  \\\n",
       "0            ENSSSCG00000000272_+_+_(72g)  ENSSSCG00000000272       False   \n",
       "1            ENSSSCG00000000272_+_+_(72g)  ENSSSCG00000000272       False   \n",
       "2            ENSSSCG00000000272_+_+_(72g)  ENSSSCG00000000272       False   \n",
       "3            ENSSSCG00000000272_+_+_(72g)  ENSSSCG00000000272       False   \n",
       "4            ENSSSCG00000000272_+_+_(72g)  ENSSSCG00000000272       False   \n",
       "..                                    ...                 ...         ...   \n",
       "15  ENSSSCG00000037647_extended_-_-_(12g)  ENSSSCG00000037647        True   \n",
       "16  ENSSSCG00000037647_extended_-_-_(12g)  ENSSSCG00000037647        True   \n",
       "17  ENSSSCG00000037647_extended_-_-_(12g)  ENSSSCG00000037647        True   \n",
       "18  ENSSSCG00000037647_extended_-_-_(12g)  ENSSSCG00000037647        True   \n",
       "19  ENSSSCG00000037647_extended_-_-_(12g)  ENSSSCG00000037647        True   \n",
       "\n",
       "                   Region                Gene  R2G_importance   R2G_rho  \\\n",
       "0   1:107949712-107950212  ENSSSCG00000004558        0.021216  0.132060   \n",
       "1   1:107659273-107659773  ENSSSCG00000004558        0.008690  0.082422   \n",
       "2   1:107689786-107690286  ENSSSCG00000004558        0.021383  0.123316   \n",
       "3   1:107886380-107886880  ENSSSCG00000004558        0.056425  0.161613   \n",
       "4   1:107949165-107949665  ENSSSCG00000004558        0.010226  0.100105   \n",
       "..                    ...                 ...             ...       ...   \n",
       "15   14:88473878-88474378  ENSSSCG00000010370        0.014041 -0.143928   \n",
       "16    5:32970064-32970564  ENSSSCG00000032595        0.016981 -0.109311   \n",
       "17    5:32981053-32981553  ENSSSCG00000032595        0.030084 -0.226317   \n",
       "18  1:163790120-163790620  ENSSSCG00000044382        0.033541 -0.113878   \n",
       "19   12:26842989-26843489  ENSSSCG00000017566        0.031242 -0.393878   \n",
       "\n",
       "    R2G_importance_x_rho  R2G_importance_x_abs_rho  TF2G_importance  \\\n",
       "0               0.002802                  0.002802         0.877556   \n",
       "1               0.000716                  0.000716         0.877556   \n",
       "2               0.002637                  0.002637         0.877556   \n",
       "3               0.009119                  0.009119         0.877556   \n",
       "4               0.001024                  0.001024         0.877556   \n",
       "..                   ...                       ...              ...   \n",
       "15             -0.002021                  0.002021         0.678860   \n",
       "16             -0.001856                  0.001856         0.506837   \n",
       "17             -0.006809                  0.006809         0.506837   \n",
       "18             -0.003820                  0.003820         0.805091   \n",
       "19             -0.012305                  0.012305         0.380228   \n",
       "\n",
       "    TF2G_regulation  TF2G_rho  TF2G_importance_x_abs_rho  \\\n",
       "0                 1  0.166144                   0.145801   \n",
       "1                 1  0.166144                   0.145801   \n",
       "2                 1  0.166144                   0.145801   \n",
       "3                 1  0.166144                   0.145801   \n",
       "4                 1  0.166144                   0.145801   \n",
       "..              ...       ...                        ...   \n",
       "15               -1 -0.162168                  -0.110089   \n",
       "16               -1 -0.149248                  -0.075644   \n",
       "17               -1 -0.149248                  -0.075644   \n",
       "18               -1 -0.107096                  -0.086222   \n",
       "19               -1 -0.207675                  -0.078964   \n",
       "\n",
       "    TF2G_importance_x_rho  \n",
       "0                0.145801  \n",
       "1                0.145801  \n",
       "2                0.145801  \n",
       "3                0.145801  \n",
       "4                0.145801  \n",
       "..                    ...  \n",
       "15               0.110089  \n",
       "16               0.075644  \n",
       "17               0.075644  \n",
       "18               0.086222  \n",
       "19               0.078964  \n",
       "\n",
       "[41413 rows x 15 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scplus_obj.uns['eRegulon_metadata']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c9bd0501-90a4-4660-ab19-9814789f5bc1",
   "metadata": {},
   "outputs": [],
   "source": [
    "def gene_convert(gene_list):\n",
    "    from gprofiler import GProfiler\n",
    "    import numpy as np\n",
    "    gp = GProfiler(return_dataframe=True)\n",
    "    x = np.array(gene_list)\n",
    "    gene_list_convert = gp.convert(organism='sscrofa', query=np.unique(x).tolist())\n",
    "    gene_list_convert['name'] = np.where(gene_list_convert['name'] == 'None', gene_list_convert['incoming'], gene_list_convert['name'])\n",
    "    return gene_list_convert"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0dc9be4b-20b3-431c-97f2-87d9a4216ee2",
   "metadata": {},
   "outputs": [],
   "source": [
    "list_tf = gene_convert(scplus_obj.uns['eRegulon_metadata']['TF'].tolist())\n",
    "list_gene = gene_convert(scplus_obj.uns['eRegulon_metadata']['Gene'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "6c61cbb1-eaba-4bfd-b754-0834d9517c28",
   "metadata": {},
   "outputs": [],
   "source": [
    "scplus_obj.uns['eRegulon_metadata']['TF'] = scplus_obj.uns['eRegulon_metadata']['TF'].replace(list_tf['incoming'].tolist(), list_tf['name'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "55588d3b-6685-4475-a9d7-1477d0ab84f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "scplus_obj.uns['eRegulon_metadata']['Gene'] = scplus_obj.uns['eRegulon_metadata']['Gene'].replace(list_gene['incoming'].tolist(), list_gene['name'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d948c08d-fe97-44ba-90a5-e1fb181bfd5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def gene_convert_part(gene_list):\n",
    "    from gprofiler import GProfiler\n",
    "    import numpy as np\n",
    "    gp = GProfiler(return_dataframe=True)\n",
    "    x = np.array(gene_list)\n",
    "    x = np.unique(x).tolist()\n",
    "    part_a = [item.split('_', 1)[0] for item in x]\n",
    "    part_convert = gp.convert(organism='sscrofa', query=part_a)\n",
    "    part_convert['name'] = np.where(part_convert['name'] == 'None', part_convert['incoming'], part_convert['name'])\n",
    "    part_b = [item.split('_', 1)[1] for item in x]\n",
    "    region_list = [f\"{itema}_{itemb}\" for itema, itemb in zip(part_convert['name'], part_b)]\n",
    "    return x, region_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "0ad45863-7b7f-428f-8cc3-51b7941d11ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "source, replacement = gene_convert_part(scplus_obj.uns['eRegulon_metadata']['Region_signature_name'].tolist())\n",
    "scplus_obj.uns['eRegulon_metadata']['Region_signature_name'] = scplus_obj.uns['eRegulon_metadata']['Region_signature_name'].replace(source, replacement)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "2c74deaf-e6f7-4994-a1b1-cd3487473996",
   "metadata": {},
   "outputs": [],
   "source": [
    "source, replacement = gene_convert_part(scplus_obj.uns['eRegulon_metadata']['Gene_signature_name'].tolist())\n",
    "scplus_obj.uns['eRegulon_metadata']['Gene_signature_name'] = scplus_obj.uns['eRegulon_metadata']['Gene_signature_name'].replace(source, replacement)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "244e4c7b-035a-4339-b58e-5207963e93aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Region_signature_name</th>\n",
       "      <th>Gene_signature_name</th>\n",
       "      <th>TF</th>\n",
       "      <th>is_extended</th>\n",
       "      <th>Region</th>\n",
       "      <th>Gene</th>\n",
       "      <th>R2G_importance</th>\n",
       "      <th>R2G_rho</th>\n",
       "      <th>R2G_importance_x_rho</th>\n",
       "      <th>R2G_importance_x_abs_rho</th>\n",
       "      <th>TF2G_importance</th>\n",
       "      <th>TF2G_regulation</th>\n",
       "      <th>TF2G_rho</th>\n",
       "      <th>TF2G_importance_x_abs_rho</th>\n",
       "      <th>TF2G_importance_x_rho</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SP1_+_+_(174r)</td>\n",
       "      <td>SP1_+_+_(72g)</td>\n",
       "      <td>SP1</td>\n",
       "      <td>False</td>\n",
       "      <td>1:107949712-107950212</td>\n",
       "      <td>CSNK1G1</td>\n",
       "      <td>0.021216</td>\n",
       "      <td>0.132060</td>\n",
       "      <td>0.002802</td>\n",
       "      <td>0.002802</td>\n",
       "      <td>0.877556</td>\n",
       "      <td>1</td>\n",
       "      <td>0.166144</td>\n",
       "      <td>0.145801</td>\n",
       "      <td>0.145801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SP1_+_+_(174r)</td>\n",
       "      <td>SP1_+_+_(72g)</td>\n",
       "      <td>SP1</td>\n",
       "      <td>False</td>\n",
       "      <td>1:107659273-107659773</td>\n",
       "      <td>CSNK1G1</td>\n",
       "      <td>0.008690</td>\n",
       "      <td>0.082422</td>\n",
       "      <td>0.000716</td>\n",
       "      <td>0.000716</td>\n",
       "      <td>0.877556</td>\n",
       "      <td>1</td>\n",
       "      <td>0.166144</td>\n",
       "      <td>0.145801</td>\n",
       "      <td>0.145801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SP1_+_+_(174r)</td>\n",
       "      <td>SP1_+_+_(72g)</td>\n",
       "      <td>SP1</td>\n",
       "      <td>False</td>\n",
       "      <td>1:107689786-107690286</td>\n",
       "      <td>CSNK1G1</td>\n",
       "      <td>0.021383</td>\n",
       "      <td>0.123316</td>\n",
       "      <td>0.002637</td>\n",
       "      <td>0.002637</td>\n",
       "      <td>0.877556</td>\n",
       "      <td>1</td>\n",
       "      <td>0.166144</td>\n",
       "      <td>0.145801</td>\n",
       "      <td>0.145801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SP1_+_+_(174r)</td>\n",
       "      <td>SP1_+_+_(72g)</td>\n",
       "      <td>SP1</td>\n",
       "      <td>False</td>\n",
       "      <td>1:107886380-107886880</td>\n",
       "      <td>CSNK1G1</td>\n",
       "      <td>0.056425</td>\n",
       "      <td>0.161613</td>\n",
       "      <td>0.009119</td>\n",
       "      <td>0.009119</td>\n",
       "      <td>0.877556</td>\n",
       "      <td>1</td>\n",
       "      <td>0.166144</td>\n",
       "      <td>0.145801</td>\n",
       "      <td>0.145801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SP1_+_+_(174r)</td>\n",
       "      <td>SP1_+_+_(72g)</td>\n",
       "      <td>SP1</td>\n",
       "      <td>False</td>\n",
       "      <td>1:107949165-107949665</td>\n",
       "      <td>CSNK1G1</td>\n",
       "      <td>0.010226</td>\n",
       "      <td>0.100105</td>\n",
       "      <td>0.001024</td>\n",
       "      <td>0.001024</td>\n",
       "      <td>0.877556</td>\n",
       "      <td>1</td>\n",
       "      <td>0.166144</td>\n",
       "      <td>0.145801</td>\n",
       "      <td>0.145801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TFDP2_extended_-_-_(14r)</td>\n",
       "      <td>TFDP2_extended_-_-_(12g)</td>\n",
       "      <td>TFDP2</td>\n",
       "      <td>True</td>\n",
       "      <td>14:88473878-88474378</td>\n",
       "      <td>ENSSSCG00000010370</td>\n",
       "      <td>0.014041</td>\n",
       "      <td>-0.143928</td>\n",
       "      <td>-0.002021</td>\n",
       "      <td>0.002021</td>\n",
       "      <td>0.678860</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.162168</td>\n",
       "      <td>-0.110089</td>\n",
       "      <td>0.110089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>TFDP2_extended_-_-_(14r)</td>\n",
       "      <td>TFDP2_extended_-_-_(12g)</td>\n",
       "      <td>TFDP2</td>\n",
       "      <td>True</td>\n",
       "      <td>5:32970064-32970564</td>\n",
       "      <td>RAP1B</td>\n",
       "      <td>0.016981</td>\n",
       "      <td>-0.109311</td>\n",
       "      <td>-0.001856</td>\n",
       "      <td>0.001856</td>\n",
       "      <td>0.506837</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.149248</td>\n",
       "      <td>-0.075644</td>\n",
       "      <td>0.075644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>TFDP2_extended_-_-_(14r)</td>\n",
       "      <td>TFDP2_extended_-_-_(12g)</td>\n",
       "      <td>TFDP2</td>\n",
       "      <td>True</td>\n",
       "      <td>5:32981053-32981553</td>\n",
       "      <td>RAP1B</td>\n",
       "      <td>0.030084</td>\n",
       "      <td>-0.226317</td>\n",
       "      <td>-0.006809</td>\n",
       "      <td>0.006809</td>\n",
       "      <td>0.506837</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.149248</td>\n",
       "      <td>-0.075644</td>\n",
       "      <td>0.075644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>TFDP2_extended_-_-_(14r)</td>\n",
       "      <td>TFDP2_extended_-_-_(12g)</td>\n",
       "      <td>TFDP2</td>\n",
       "      <td>True</td>\n",
       "      <td>1:163790120-163790620</td>\n",
       "      <td>ENSSSCG00000044382</td>\n",
       "      <td>0.033541</td>\n",
       "      <td>-0.113878</td>\n",
       "      <td>-0.003820</td>\n",
       "      <td>0.003820</td>\n",
       "      <td>0.805091</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.107096</td>\n",
       "      <td>-0.086222</td>\n",
       "      <td>0.086222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>TFDP2_extended_-_-_(14r)</td>\n",
       "      <td>TFDP2_extended_-_-_(12g)</td>\n",
       "      <td>TFDP2</td>\n",
       "      <td>True</td>\n",
       "      <td>12:26842989-26843489</td>\n",
       "      <td>ACSF2</td>\n",
       "      <td>0.031242</td>\n",
       "      <td>-0.393878</td>\n",
       "      <td>-0.012305</td>\n",
       "      <td>0.012305</td>\n",
       "      <td>0.380228</td>\n",
       "      <td>-1</td>\n",
       "      <td>-0.207675</td>\n",
       "      <td>-0.078964</td>\n",
       "      <td>0.078964</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41413 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Region_signature_name       Gene_signature_name     TF is_extended  \\\n",
       "0             SP1_+_+_(174r)             SP1_+_+_(72g)    SP1       False   \n",
       "1             SP1_+_+_(174r)             SP1_+_+_(72g)    SP1       False   \n",
       "2             SP1_+_+_(174r)             SP1_+_+_(72g)    SP1       False   \n",
       "3             SP1_+_+_(174r)             SP1_+_+_(72g)    SP1       False   \n",
       "4             SP1_+_+_(174r)             SP1_+_+_(72g)    SP1       False   \n",
       "..                       ...                       ...    ...         ...   \n",
       "15  TFDP2_extended_-_-_(14r)  TFDP2_extended_-_-_(12g)  TFDP2        True   \n",
       "16  TFDP2_extended_-_-_(14r)  TFDP2_extended_-_-_(12g)  TFDP2        True   \n",
       "17  TFDP2_extended_-_-_(14r)  TFDP2_extended_-_-_(12g)  TFDP2        True   \n",
       "18  TFDP2_extended_-_-_(14r)  TFDP2_extended_-_-_(12g)  TFDP2        True   \n",
       "19  TFDP2_extended_-_-_(14r)  TFDP2_extended_-_-_(12g)  TFDP2        True   \n",
       "\n",
       "                   Region                Gene  R2G_importance   R2G_rho  \\\n",
       "0   1:107949712-107950212             CSNK1G1        0.021216  0.132060   \n",
       "1   1:107659273-107659773             CSNK1G1        0.008690  0.082422   \n",
       "2   1:107689786-107690286             CSNK1G1        0.021383  0.123316   \n",
       "3   1:107886380-107886880             CSNK1G1        0.056425  0.161613   \n",
       "4   1:107949165-107949665             CSNK1G1        0.010226  0.100105   \n",
       "..                    ...                 ...             ...       ...   \n",
       "15   14:88473878-88474378  ENSSSCG00000010370        0.014041 -0.143928   \n",
       "16    5:32970064-32970564               RAP1B        0.016981 -0.109311   \n",
       "17    5:32981053-32981553               RAP1B        0.030084 -0.226317   \n",
       "18  1:163790120-163790620  ENSSSCG00000044382        0.033541 -0.113878   \n",
       "19   12:26842989-26843489               ACSF2        0.031242 -0.393878   \n",
       "\n",
       "    R2G_importance_x_rho  R2G_importance_x_abs_rho  TF2G_importance  \\\n",
       "0               0.002802                  0.002802         0.877556   \n",
       "1               0.000716                  0.000716         0.877556   \n",
       "2               0.002637                  0.002637         0.877556   \n",
       "3               0.009119                  0.009119         0.877556   \n",
       "4               0.001024                  0.001024         0.877556   \n",
       "..                   ...                       ...              ...   \n",
       "15             -0.002021                  0.002021         0.678860   \n",
       "16             -0.001856                  0.001856         0.506837   \n",
       "17             -0.006809                  0.006809         0.506837   \n",
       "18             -0.003820                  0.003820         0.805091   \n",
       "19             -0.012305                  0.012305         0.380228   \n",
       "\n",
       "    TF2G_regulation  TF2G_rho  TF2G_importance_x_abs_rho  \\\n",
       "0                 1  0.166144                   0.145801   \n",
       "1                 1  0.166144                   0.145801   \n",
       "2                 1  0.166144                   0.145801   \n",
       "3                 1  0.166144                   0.145801   \n",
       "4                 1  0.166144                   0.145801   \n",
       "..              ...       ...                        ...   \n",
       "15               -1 -0.162168                  -0.110089   \n",
       "16               -1 -0.149248                  -0.075644   \n",
       "17               -1 -0.149248                  -0.075644   \n",
       "18               -1 -0.107096                  -0.086222   \n",
       "19               -1 -0.207675                  -0.078964   \n",
       "\n",
       "    TF2G_importance_x_rho  \n",
       "0                0.145801  \n",
       "1                0.145801  \n",
       "2                0.145801  \n",
       "3                0.145801  \n",
       "4                0.145801  \n",
       "..                    ...  \n",
       "15               0.110089  \n",
       "16               0.075644  \n",
       "17               0.075644  \n",
       "18               0.086222  \n",
       "19               0.078964  \n",
       "\n",
       "[41413 rows x 15 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scplus_obj.uns['eRegulon_metadata']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "bbd5b0af-49f6-480d-a4f8-5d4396ced8a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "list_gene = gene_convert(scplus_obj.metadata_genes['gene_ids'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "b624cb97-5d5f-48b5-acdd-847af87f1bf8",
   "metadata": {},
   "outputs": [],
   "source": [
    "scplus_obj.metadata_genes['gene_ids'] = scplus_obj.metadata_genes['gene_ids'].replace(list_gene['incoming'].tolist(), list_gene['name'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "7abc04c4-2c94-463f-a92a-b346e4d1ec16",
   "metadata": {},
   "outputs": [],
   "source": [
    "from numpy import copy\n",
    "newArray = copy(scplus_obj.metadata_genes.index.values)\n",
    "for k, v in zip(list_gene['incoming'].tolist(), list_gene['name'].tolist()): scplus_obj.metadata_genes.index.values[scplus_obj.metadata_genes.index.values==k] = v"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e1b4d279-15ae-4588-8f75-43988b236d6a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_ids</th>\n",
       "      <th>feature_types</th>\n",
       "      <th>genome</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_ids</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSSSCG00000048769</th>\n",
       "      <td>ENSSSCG00000048769</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PSMB1</th>\n",
       "      <td>PSMB1</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DLL1</th>\n",
       "      <td>DLL1</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DYNLT2</th>\n",
       "      <td>DYNLT2</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PHF10</th>\n",
       "      <td>PHF10</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSSSCG00000036928</th>\n",
       "      <td>ENSSSCG00000036928</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSSSCG00000040795</th>\n",
       "      <td>ENSSSCG00000040795</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSSSCG00000023819</th>\n",
       "      <td>ENSSSCG00000023819</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRY</th>\n",
       "      <td>SRY</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSSSCG00000034141</th>\n",
       "      <td>ENSSSCG00000034141</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>sus_scrofa_11_ensembl_arc_108</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>35670 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              gene_ids    feature_types  \\\n",
       "gene_ids                                                  \n",
       "ENSSSCG00000048769  ENSSSCG00000048769  Gene Expression   \n",
       "PSMB1                            PSMB1  Gene Expression   \n",
       "DLL1                              DLL1  Gene Expression   \n",
       "DYNLT2                          DYNLT2  Gene Expression   \n",
       "PHF10                            PHF10  Gene Expression   \n",
       "...                                ...              ...   \n",
       "ENSSSCG00000036928  ENSSSCG00000036928  Gene Expression   \n",
       "ENSSSCG00000040795  ENSSSCG00000040795  Gene Expression   \n",
       "ENSSSCG00000023819  ENSSSCG00000023819  Gene Expression   \n",
       "SRY                                SRY  Gene Expression   \n",
       "ENSSSCG00000034141  ENSSSCG00000034141  Gene Expression   \n",
       "\n",
       "                                           genome  \n",
       "gene_ids                                           \n",
       "ENSSSCG00000048769  sus_scrofa_11_ensembl_arc_108  \n",
       "PSMB1               sus_scrofa_11_ensembl_arc_108  \n",
       "DLL1                sus_scrofa_11_ensembl_arc_108  \n",
       "DYNLT2              sus_scrofa_11_ensembl_arc_108  \n",
       "PHF10               sus_scrofa_11_ensembl_arc_108  \n",
       "...                                           ...  \n",
       "ENSSSCG00000036928  sus_scrofa_11_ensembl_arc_108  \n",
       "ENSSSCG00000040795  sus_scrofa_11_ensembl_arc_108  \n",
       "ENSSSCG00000023819  sus_scrofa_11_ensembl_arc_108  \n",
       "SRY                 sus_scrofa_11_ensembl_arc_108  \n",
       "ENSSSCG00000034141  sus_scrofa_11_ensembl_arc_108  \n",
       "\n",
       "[35670 rows x 3 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scplus_obj.metadata_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "05f0041d-d0b6-48a1-919f-9953d36a0fce",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Format eRegulons\n",
    "from scenicplus.eregulon_enrichment import *\n",
    "get_eRegulons_as_signatures(scplus_obj, eRegulon_metadata_key='eRegulon_metadata', key_added='eRegulon_signatures')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "905708e6-e801-4324-8010-8ecc2b099c4f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ| 241/241 [00:25<00:00,  9.62it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5.779541083176931\n"
     ]
    }
   ],
   "source": [
    "## Score chromatin layer\n",
    "# Region based raking\n",
    "from scenicplus.cistromes import *\n",
    "import time\n",
    "start_time = time.time()\n",
    "region_ranking = make_rankings(scplus_obj, target='region')\n",
    "# Score region regulons\n",
    "score_eRegulons(scplus_obj,\n",
    "                ranking = region_ranking,\n",
    "                eRegulon_signatures_key = 'eRegulon_signatures',\n",
    "                key_added = 'eRegulon_AUC',\n",
    "                enrichment_type= 'region',\n",
    "                auc_threshold = 0.05,\n",
    "                normalize = False,\n",
    "                n_cpu = 1)\n",
    "time = time.time()-start_time\n",
    "print(time/60)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f83d21c4-ab8e-46f9-8c7c-2371a1763cfd",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ| 241/241 [00:15<00:00, 15.93it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.1655084490776062\n"
     ]
    }
   ],
   "source": [
    "## Score transcriptome layer\n",
    "# Gene based raking\n",
    "from scenicplus.cistromes import *\n",
    "import time\n",
    "start_time = time.time()\n",
    "gene_ranking = make_rankings(scplus_obj, target='gene')\n",
    "# Score gene regulons\n",
    "score_eRegulons(scplus_obj,\n",
    "                gene_ranking,\n",
    "                eRegulon_signatures_key = 'eRegulon_signatures',\n",
    "                key_added = 'eRegulon_AUC',\n",
    "                enrichment_type = 'gene',\n",
    "                auc_threshold = 0.05,\n",
    "                normalize= False,\n",
    "                n_cpu = 1)\n",
    "time = time.time()-start_time\n",
    "print(time/60)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "db0d24f6-7365-48a7-ae7e-786418130060",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.43292161226272585\n"
     ]
    }
   ],
   "source": [
    "# Generate pseudobulks\n",
    "import time\n",
    "start_time = time.time()\n",
    "generate_pseudobulks(scplus_obj,\n",
    "                         variable = 'GEX_Clusters_Time',\n",
    "                         auc_key = 'eRegulon_AUC',\n",
    "                         signature_key = 'Gene_based',\n",
    "                         nr_cells = 1,\n",
    "                         nr_pseudobulks = 100,\n",
    "                         seed=555)\n",
    "generate_pseudobulks(scplus_obj,\n",
    "                         variable = 'GEX_Clusters_Time',\n",
    "                         auc_key = 'eRegulon_AUC',\n",
    "                         signature_key = 'Region_based',\n",
    "                         nr_cells = 1,\n",
    "                         nr_pseudobulks = 100,\n",
    "                         seed=555)\n",
    "time = time.time()-start_time\n",
    "print(time/60)"
   ]
  },
  {
   "cell_type": "raw",
   "id": "348963bf-f5d2-4faa-b32e-57521181f232",
   "metadata": {},
   "source": [
    "%matplotlib inline\n",
    "import seaborn as sns\n",
    "sns.set_style(\"white\")\n",
    "colors = [\"#E9842C\",\"#F8766D\", \"#BC9D00\", \"#00C0B4\", \"#9CA700\", \"#6FB000\", \"#00B813\", \"#00BD61\", \"#00C08E\", \"#00BDD4\",\n",
    "           \"#00A7FF\", \"#7F96FF\", \"#E26EF7\", \"#FF62BF\", \"#D69100\", \"#BC81FF\"]\n",
    "categories = sorted(set(scplus_obj.metadata_cell['GEX_Clusters']))\n",
    "color_dict = dict(zip(categories, colors[0:len(categories)]))\n",
    "prune_plot(scplus_obj,\n",
    "           'NANOG_+_+',\n",
    "           pseudobulk_variable = 'GEX_Clusters',\n",
    "           show_dot_plot = True,\n",
    "           show_line_plot = False,\n",
    "           color_dict = color_dict,\n",
    "           use_pseudobulk = True,\n",
    "           auc_key = 'eRegulon_AUC',\n",
    "           signature_key = 'Region_based',\n",
    "           seed=555)"
   ]
  },
  {
   "cell_type": "raw",
   "id": "9541bf7a-19c4-4325-ace7-53a56768273c",
   "metadata": {},
   "source": [
    "# Gene based\n",
    "%matplotlib inline\n",
    "sns.set_style(\"white\")\n",
    "colors = [\"#E9842C\",\"#F8766D\", \"#BC9D00\", \"#00C0B4\", \"#9CA700\", \"#6FB000\", \"#00B813\", \"#00BD61\", \"#00C08E\", \"#00BDD4\",\n",
    "           \"#00A7FF\", \"#7F96FF\", \"#E26EF7\", \"#FF62BF\", \"#D69100\", \"#BC81FF\"]\n",
    "categories = sorted(set(scplus_obj.metadata_cell['GEX_Clusters']))\n",
    "color_dict = dict(zip(categories, colors[0:len(categories)]))\n",
    "prune_plot(scplus_obj,\n",
    "           'NANOG_+_+',\n",
    "           pseudobulk_variable = 'GEX_Clusters',\n",
    "           show_dot_plot = True,\n",
    "           show_line_plot = False,\n",
    "           color_dict = color_dict,\n",
    "           use_pseudobulk = True,\n",
    "           auc_key = 'eRegulon_AUC',\n",
    "           signature_key = 'Gene_based',\n",
    "           seed=555)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "3711dbaa-9169-4135-b4e9-7c319ef4626d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correlation between region based regulons and gene based regulons\n",
    "import pandas\n",
    "df1 = scplus_obj.uns['eRegulon_AUC']['Gene_based'].copy()\n",
    "df2 = scplus_obj.uns['eRegulon_AUC']['Region_based'].copy()\n",
    "df1.columns = [x.split('_(')[0] for x in df1.columns]\n",
    "df2.columns = [x.split('_(')[0] for x in df2.columns]\n",
    "correlations = df1.corrwith(df2, axis = 0)\n",
    "correlations = correlations[abs(correlations) > 0.2]\n",
    "# Kepp only R2G +\n",
    "keep = [x for x in correlations.index if '+_+' in x] + [x for x in correlations.index if '-_+' in x]\n",
    "# Keep extended if not direct\n",
    "extended = [x for x in keep if 'extended' in x]\n",
    "direct = [x for x in keep if not 'extended' in x]\n",
    "keep_extended = [x for x in extended if not x.replace('extended_', '') in direct]\n",
    "keep = direct + keep_extended\n",
    "# Keep regulons with more than 10 genes\n",
    "keep_gene = [x for x in scplus_obj.uns['eRegulon_AUC']['Gene_based'].columns if x.split('_(')[0] in keep]\n",
    "keep_gene = [x for x in keep_gene if (int(x.split('_(')[1].replace('g)', '')) > 10)]\n",
    "keep_all = [x.split('_(')[0] for x in keep_gene]\n",
    "keep_region = [x for x in scplus_obj.uns['eRegulon_AUC']['Region_based'].columns if x.split('_(')[0] in keep]\n",
    "scplus_obj.uns['selected_eRegulons'] = {}\n",
    "scplus_obj.uns['selected_eRegulons']['Gene_based'] = keep_gene\n",
    "scplus_obj.uns['selected_eRegulons']['Region_based'] = keep_region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "5bd0bd80-0d4d-49c8-903a-7a74a93fe31f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import dill as dill"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "990152d8-d6ce-4ef7-a024-cee4066b5631",
   "metadata": {},
   "outputs": [],
   "source": [
    "dill.dump(scplus_obj, open('/home/adufour/work/scenic_omics/embryos/SCENIC/scplus_obj.pkl', 'wb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "43cdf888-4bb0-4e1e-93ff-b819d924fab0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:scenicplus]",
   "language": "python",
   "name": "conda-env-scenicplus-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.17"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
